smcore/version.cpp

```diff
@@ -81,11 +82,28 @@ namespace smos
             return result;
         }
         //******************************************************************************
-        void Version::SetVersion(const short &major, const short &minor, const short &revision)
+        void Version::SetVersion(const short major, const short minor, const short revision)
         {
             this->m_major = major;
             this->m_minor = minor;
             this->m_revision = revision;
         }
+        //******************************************************************************
+        std::ostream &operator<<(std::ostream &os, const Version &obj)
+        {
+            os << obj.m_major;
```

Comment:
I'm not sure if the "short" type is portable - is it always the same size? Maybe it would be better to use int16_t or int8_t or even better uint16_t or uint8_t.

Reply:
You're absolutely right, I'll change this.